### DISCUSSION TITLE
Which attributes are selectable for Virtual Category rules?

### DISCUSSION THREAD

**Discussion by amenk** [Smile-SA/elasticsuite#2384] [Q&A] [answered]
We did not find this information on https://github.com/Smile-SA/elasticsuite/wiki/VirtualCategories

Our quess was it's use-in-promo rules, but that is not the case.

**Comment by rbayet** [comment, marked as answer]
Hello @amenk,

Indeed the information is missing in the wiki but all compatible attributes that are in the mapping (that is, indexed) are supposed to be available in catalog rules (optimizers and virtual categories).
That includes attributes with "Used in Promo Rules = Yes" only attributes (so for instance, you're able to display or boost products with a good margin).

Depending on your version of Elasticsuite (<2.10.6), however, you might need to perform a full reindex (so the mapping is refreshed) for an attribute that you just added to the index by putting it as "Used in Promo Rules = Yes" to pop up.

Regards,

**Reply by amenk** [reply]
We are at 2.10.4 and a full reindex fixed the problem. But it does not have to do with "Use for Promo Rule Conditions" -> also attributes where this is "No" are displayed (which is just fine)

**Reply by rbayet** [reply]
> [...] But it does not have to do with "Use for Promo Rule Conditions" -> also attributes where this is "No" are displayed (which is just fine)

Hence why I said "all compatible attributes that are in the mapping (that is, indexed)", so that includes attributes (filterable OR filterable in search OR searchable OR used in promo rules).
Logically that should include sortable attributes too.

**Reply by amenk** [reply]
Now I got i, thanks. Shall we add this to the wiki?

**Reply by romainruaud** [reply]
Yes, I think that should be added in the wiki for everyone's knowledge, I don't know if you are able to contribute to the wiki or not btw.

**Reply by amenk** [reply]
I think I can't
